lib/monaco-adapter.js

```diff
@@ -0,0 +1,298 @@
+firepad.MonacoAdapter = (function() {
+  MonacoAdapter.prototype.ignoreChanges = false;
+
+  function MonacoAdapter(monacoInstance) {
+    this.onCursorActivity = __bind(this.onCursorActivity, this);
+    this.onFocus = __bind(this.onFocus, this);
+    this.onBlur = __bind(this.onBlur, this);
+    this.onChange = __bind(this.onChange, this);
+    this.monaco = monacoInstance;
+    this.monacoModel = this.monaco.getModel();
+
+    this.grabDocumentState();
+    this.monaco.onDidChangeModelContent(this.onChange);
+    this.monaco.onDidBlurEditor(this.onBlur);
+    this.monaco.onDidFocusEditor(this.onFocus);
+    this.monaco.onDidChangeCursorPosition(this.onCursorActivity);
+  }
+
+  MonacoAdapter.prototype.grabDocumentState = function() {
+    this.lastDocLines = this.monacoModel.getLinesContent();
+    var range = this.monaco.getSelection();
+    return this.lastCursorRange = range;
+  };
+
+  MonacoAdapter.prototype.detach = function() {
+    this.monaco.onDidChangeModelContent(this.onChange).dispose();
```

Comment:
I haven't dug into Monaco's implementation, but I kinda' suspect that this is attaching a _new_ listener and then immediately removing it, and likely isn't removing the listener you attached in the constructor.

Reply:
Yeah! Even I'm not that sure about this. A better addition would be to have
`this.onChange.dispose()` this will call the dispose of for the actual attached `onChange`.